### PR TITLE
DEX 461 & 460 add more encounter and individual data to API returns

### PR DESCRIPTION
**Pull request by colinwkingen**
It just made sense to mash these tickets together. The individual data is added to the return in the EDM when max detail level is requested.

Returned from Sighting API:
```javascript

{
	'comments': 'None',
	'encounters': [{
		'individual': {
			'names': {
				'nickname': 'Meatloaf',
				'defaultName': 'Micheal Aday'
			},
			'id': 'fc35b212-3d96-4071-b290-6ab22df4ec0b'
		},
		'customFields': {},
		'locationId': 'Saturn',
		'decimalLongitude': '25.9999',
		'decimalLatitude': '25.9999',
		'verbatimLocality': 'Saturn',
		'id': 'f9aabbd7-56c6-4a68-a097-46fc5a9e8697',
		'time': '2010-01-01T01:01:01Z',
		'version': 1633473655066,
		'hasView': True,
		'createdHouston': '2021-10-05T22:40:55.091886+00:00',
		'guid': 'f9aabbd7-56c6-4a68-a097-46fc5a9e8697',
		'submitter': {
			'guid': 'e5435041-b65d-41d7-9a5a-a3168d30ce28',
			'full_name': 'First Middle Last',
			'profile_fileupload': None
		},
		'owner': {
			'guid': 'e5435041-b65d-41d7-9a5a-a3168d30ce28',
			'full_name': 'First Middle Last',
			'profile_fileupload': None
		},
		'hasEdit': True,
		'updatedHouston': '2021-10-05T22:40:55.091905+00:00'
	}],
	'createdEDM': '2021-10-05 22:40:55',
	'customFields': {},
	'locationId': 'test',
	'decimalLongitude': 25.9999,
	'startTime': '2021-10-05T22:40:44.692Z',
	'decimalLatitude': 25.9999,
	'verbatimLocality': 'Saturn',
	'id': 'c3c11028-991c-43aa-be38-45f0dd922415',
	'encounterCounts': {
		'lifeStage': {},
		'sex': {},
		'individuals': 1
	},
	'version': 1633473655066,
	'featuredAssetGuid': None,
	'hasView': True,
	'createdHouston': '2021-10-05T22:40:55.084494+00:00',
	'guid': 'c3c11028-991c-43aa-be38-45f0dd922415',
	'assets': [],
	'hasEdit': True,
	'updatedHouston': '2021-10-05T22:40:55.084518+00:00'
}

```

### CODE REVIEW

**Comment by brmscheiner** [comment]
lgtm, the only thing i notice is that ID is there twice (`id` and `guid`)

File: app/modules/individuals/schemas.py [hunk 1]
```diff
@@ -43,3 +43,14 @@ class Meta(BaseIndividualSchema.Meta):
             Individual.created.key,
             Individual.updated.key,
         )
+
+
+class SightingApiIndividualSchema(BaseIndividualSchema):
+    hasView = base_fields.Function(Individual.current_user_has_view_permission)
```
Comment (hwindsor) [inline app/modules/individuals/schemas.py:49]: hasView and hasEdit (implementations) should really be moved into the BaseIndividualSchema, as they are in BaseEncounterSchema and then can remove this duplication in the derived Schema classes.

File: tests/modules/individuals/resources/test_modify_individual.py [hunk 2]
```diff
@@ -12,36 +12,49 @@ def test_modify_individual_edm_fields(db, flask_app_client, researcher_1):
...
-        assert response_json['result']['encounters'][0]['id']
+        assert response_json['encounters']
+        assert response_json['encounters'][0]['id']
 
-        guid = response_json['result']['encounters'][0]['id']
+        guid = response_json['encounters'][0]['id']
         enc = Encounter.query.get(guid)
         assert enc is not None
 
         with db.session.begin():
```
Comment (hwindsor) [inline tests/modules/individuals/resources/test_modify_individual.py:44]: same question as above, why is this needed?
